### PR TITLE
PLANET-3581 Code cleanup: Dynamic SQL Query

### PR DESCRIPTION
**Pull request by dpivo**
Dynamic SQL construction in exporter.php replaced with prepared statement.

### CODE REVIEW

**Comment by comzeradd** [comment]
@dpivo can you rebase on `release/v1.43.0`, so we can make sure that the tests will pass?

Here is a [rebase example](https://planet4.greenpeace.org/handbook/dev-git-best-practices/#rebase) with the develop branch, that you can follow.

**Comment by comzeradd** [comment]
Tests are still failing but not due to this PR.

File: exporter.php [hunk 1]
```diff
@@ -221,7 +221,7 @@ function p4_px_single_post_authors_list( $post_ids ) {
 	$authors = array();
 
 	// Ignore lint on the following line because it doesn't detect the context of the $post_ids.
-	$results = $wpdb->get_results( sprintf( "SELECT DISTINCT post_author FROM $wpdb->posts WHERE ID IN( %s ) AND post_status != 'auto-draft'", $post_ids ) ); // phpcs:ignore
+	$results = $wpdb->get_results( $wpdb->prepare( "SELECT DISTINCT post_author FROM $wpdb->posts WHERE ID IN( %s ) AND post_status != 'auto-draft'", $post_ids ) ); // phpcs:ignore
```
Comment (sagarsdeshmukh) [inline exporter.php]: @dpivo - This query will give incorrect results. Did you test the query output before and after change?

Comment (dpivo) [reply exporter.php]: I only tested it after I implemented it. I will test before and after.

**Pull request merged**